DAOS-12429 build: Enable coredumps by default

brianjmurrell opened the pull request:
If the site admin or distro has not set up any other means of
collecting core files.

### Before requesting gatekeeper:

* [ ] Two review approvals and any prior change requests have been resolved.
* [ ] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [ ] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [ ] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [ ] Any tests skipped by the ticket being addressed have been run and passed in the PR.

### Gatekeeper:

* [ ] You are the appropriate gatekeeper to be landing the patch.
* [ ] The PR has 2 reviews by people familiar with the code, including appropriate watchers.
* [ ] Githooks were used. If not, request that user install them and check copyright dates.
* [ ] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [ ] All builds have passed.  Check non-required builds for any new compiler warnings.
* [ ] Sufficient testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [ ] If applicable, the PR has addressed any potential version compatibility issues.
* [ ] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [ ] Extra checks if forced landing is requested
  * [ ] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [ ] No new NLT or valgrind warnings.  Check the classic view.
  * [ ] Quick-build or Quick-functional is not used.
* [ ] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

brianjmurrell committed:
DAOS-12429 build: Enable coredumps by default

If the site admin or distro has not set up any other means of
collecting core files.

Use coredumpctl in handling core dumps from systemd-coredump.

Quick-Functional: true
PR-repos: daos@PR-11447:15
RPM-test-version: 2.3.103
Skip-coverity-test: true
Skip-func-test-leap15: false
Skip-func-hw-test: true
Skip-scan-rpms: true
Skip-test-rpms: true
Test-tag: test_core_files
Skip-PR-comments: true

Required-githooks: true


# NOTE: The following default set of pragmas will result in no build
#       or test.
#
#       Please edit the list of pragmas to produce the minimal amount of build
#       and test to prove your PR works.
#
#       Or if you have already done the above, and your PR is ready for it's
#       final for-landing run, remove all of the following pragmas.
#
#       More detail can be found at
#       https://wiki.hpdd.intel.com/display/CI/Commit+Pragmas.
Skip-checkpatch: true
Skip-python-bandit: true
Skip-build: true
# Or you can skip specific PR build stages:
#Skip-build-ubuntu20-rpm: true
#Skip-build-el8-rpm: true
#Skip-build-leap15-rpm: true
#Skip-build-el8-gcc: true
#Skip-build-el8-gcc-debug: true
#Skip-build-el8-gcc-release: true
#Skip-build-leap15-gcc: true
#Skip-build-leap15-icc: true
#Skip-build-ubuntu-clang: true

# And/or you could just enable quick-build below which reduces build tims considerably
Quick-build: true
# Just build and run Functional tests, skipping all others
Quick-Functional: true
# or building with multiple jobs
#Parallel-build: true

Skip-unit-tests: true
# Or you could choose which Unit Test stages to skip below
#Skip-nlt: true
#Skip-unit-test: true
#Skip-unit-test-memcheck: true
Skip-test: true

# Every run should include the features it is testing.  Uncomment the below
# line and add the features (space separated) your PR should be testing for.
# I.e. rebuild, ...
#Features: 

# Or you can choose which test stages to skip below
#Skip-coverity-test: true
# Skip all Functional test stages
#Skip-func-test: true
# Skip all Functional test stages on VMs
#Skip-func-test-vm: true
# Don't skip the valgrind functional test
#Skip-func-test-vm-valgrind: false
# Skip Functional test stages on EL7, EL8 or Leap 15
#Skip-func-test-el7: true
#Skip-func-test-el8: true
#Skip-func-test-leap15: true
# Skip all Functional test stages on hardware
#Skip-func-hw-test: true
# or just specific hardware test stages
#Skip-func-hw-test-small: true
#Skip-func-hw-test-medium: true
#Skip-func-hw-test-large: true
#Skip-scan-rpms: true
#Skip-test-rpms: true
# or per distro variants:
#Skip-scan-centos-rpms: true
#Skip-scan-centos-7-rpms: true
#Skip-scan-centos-8-rpms: true
#Skip-scan-leap-15-rpms: true
#Skip-test-centos-rpms: true
#Skip-test-centos-7-rpms: true
#Skip-test-centos-8-rpms: true
#Skip-test-centos-8.3-rpms: true

# If you do run any of the above Functional test stages, please use the following to
# limit your tests run to the tests/features you are working on
# You can add the tags for your tests to the appropriate pragmas below:
#Test-tag: -hw
#Test-tag-hw-small: hw,small
#Test-tag-hw-medium: hw,medium,ib2
#Test-tag-hw-large: hw,large

# If you want to allow Hardware testing to run even if VM testing fails:
Allow-unstable-test: true

# if you want to test your PR with a PR of a component, you can speify that PR:
#PR-repos: project@PR-number[:build_num]

# if you want to disable using RPMs in the Functional tests:
#RPM-test: false

# Or if you want to test with daos RPMs that are already built (i.e. in another PR or in the repo):
#RPM-test-version: version[-release]
# to get the latest RPMs for the current verison:
#RPM-test-version: 2.3.103

# This should be figured out automatically, but you can force it with
#Doc-only: true

# If you'd prefer not to have your PRs cluttered up with stage failure
# comments, uncomment this
# Skip-PR-comments: true

# ansible-role-snapshot_host-branch: branch-name
# ansible-role-kvm-branch: branch-name

# to use a different cluster for various stages
# VM1-label: ci_vm1
# Ubuntu-VM9-label: ci_vm9
# Leap15-VM9-label: ci_vm9
# EL8-VM9-label: ci_vm9

# to tell CI not to skip a test due to a ticket
# Fixes: DAOS-1234
#
# If you are ready to push a full CI build and test you can uncomment
# the below pragma:
#Full-CI-run: true
Signed-off-by: Brian J. Murrell <brian.murrell@intel.com>

phender commented on src/tests/ftest/harness/core_files.py:
```diff
@@ -45,15 +45,16 @@ def test_core_files(self):
-                local_core_file.write("THIS IS JUST A TEST\n")
-        except IOError as error:
-            self.fail("Error writing {}: {}".format(local_core_file, str(error)))
+        if not results.stdout.startswith('|'):
```
Instead of this should line 45 run the new `scripts/get_core_file_pattern.sh` script?  This makes the test a little less deterministic, as it currently always produces a fake core.gdb file to confirm that launch.py handles it correctly.

brianjmurrell replied:
Good idea!

phender commented on src/tests/ftest/launch.py:
```diff
@@ -2202,6 +2194,15 @@ def process(self, test, repeat, stop_daos, archive, rename, jenkinslog, core_fil
+            # Get the core file pattern information
+            try:
+                core_files = self._get_core_file_pattern(
+                    test.host_info.servers.hosts, test.host_info.servers.hosts, process_cores)
```
We should be checking all hosts for core files not just the hosts currently selected to act in the server role:
```suggestion
                    test.host_info.servers.hosts, test.host_info.clients.hosts, process_cores)
```

brianjmurrell replied:
Done

phender commented on src/tests/ftest/launch.py:
```diff
@@ -2202,6 +2194,15 @@ def process(self, test, repeat, stop_daos, archive, rename, jenkinslog, core_fil
+                    test.host_info.servers.hosts, test.host_info.servers.hosts, process_cores)
+            except LaunchException:
+                message = "Error obtaining the core file pattern information"
+                return self.get_exit_status(1, message, "Setup", sys.exc_info())
```
Since we're in the process of running each test file and processing the results, we don't want to exit launch if we we encounter this error.  Instead we should mark the execution of the test file as failed:
```suggestion
                self._fail_test(self.result.tests[-1], "Process", message, sys.exc_info())
                return_code |= 256
                core_files = {}
```

brianjmurrell replied:
Done.

brianjmurrell committed:
Update

Required-githooks: true


#Pragmas from previous commit message:
Quick-Functional: true
PR-repos: daos@PR-11447:15
RPM-test-version: 2.3.103
Skip-coverity-test: true
Skip-func-test-leap15: false
Skip-func-hw-test: true
Skip-scan-rpms: true
Skip-test-rpms: true
Test-tag: test_core_files
Skip-PR-comments: true
Skip-checkpatch: true
Skip-python-bandit: true
Skip-build: true
Quick-build: true
Quick-Functional: true
Skip-unit-tests: true
Skip-test: true
Allow-unstable-test: true
#RPM-test-version: version[-release]
#RPM-test-version: 2.3.103
# VM1-label: ci_vm1
# Ubuntu-VM9-label: ci_vm9
# Leap15-VM9-label: ci_vm9
# EL8-VM9-label: ci_vm9
Signed-off-by: Brian J. Murrell <brian.murrell@intel.com>
Quick-Functional: true
Skip-coverity-test: true
Skip-func-test-leap15: false
Skip-func-hw-test: true
Skip-scan-rpms: true
Skip-test-rpms: true
Test-tag: test_core_files
Skip-PR-comments: true

brianjmurrell committed:
Fixy python script perms


#Pragmas from previous commit message:
Quick-Functional: true
PR-repos: daos@PR-11447:15
RPM-test-version: 2.3.103
Skip-coverity-test: true
Skip-func-test-leap15: false
Skip-func-hw-test: true
Skip-scan-rpms: true
Skip-test-rpms: true
Test-tag: test_core_files
Skip-PR-comments: true
Skip-checkpatch: true
Skip-python-bandit: true
Skip-build: true
Quick-build: true
Quick-Functional: true
Skip-unit-tests: true
Skip-test: true
Allow-unstable-test: true
#RPM-test-version: version[-release]
#RPM-test-version: 2.3.103
# VM1-label: ci_vm1
# Ubuntu-VM9-label: ci_vm9
# Leap15-VM9-label: ci_vm9
# EL8-VM9-label: ci_vm9
Quick-Functional: true
Skip-coverity-test: true
Skip-func-test-leap15: false
Skip-func-hw-test: true
Skip-scan-rpms: true
Skip-test-rpms: true
Test-tag: test_core_files
Skip-PR-comments: true
Signed-off-by: Brian J. Murrell <brian.murrell@intel.com>
Quick-Functional: true
Skip-coverity-test: true
Skip-func-test-leap15: false
Skip-func-hw-test: true
Skip-scan-rpms: true
Skip-test-rpms: true
Test-tag: test_core_files
Skip-PR-comments: true

brianjmurrell committed:
Updates


#Pragmas from previous commit message:
Quick-Functional: true
PR-repos: daos@PR-11447:15
RPM-test-version: 2.3.103
Skip-coverity-test: true
Skip-func-test-leap15: false
Skip-func-hw-test: true
Skip-scan-rpms: true
Skip-test-rpms: true
Test-tag: test_core_files
Skip-PR-comments: true
Skip-checkpatch: true
Skip-python-bandit: true
Skip-build: true
Quick-build: true
Quick-Functional: true
Skip-unit-tests: true
Skip-test: true
Allow-unstable-test: true
#RPM-test-version: version[-release]
#RPM-test-version: 2.3.103
# VM1-label: ci_vm1
# Ubuntu-VM9-label: ci_vm9
# Leap15-VM9-label: ci_vm9
# EL8-VM9-label: ci_vm9
Quick-Functional: true
Skip-coverity-test: true
Skip-func-test-leap15: false
Skip-func-hw-test: true
Skip-scan-rpms: true
Skip-test-rpms: true
Test-tag: test_core_files
Skip-PR-comments: true
Quick-Functional: true
Skip-coverity-test: true
Skip-func-test-leap15: false
Skip-func-hw-test: true
Skip-scan-rpms: true
Skip-test-rpms: true
Test-tag: test_core_files
Skip-PR-comments: true
Signed-off-by: Brian J. Murrell <brian.murrell@intel.com>
Quick-Functional: true
Skip-coverity-test: true
Skip-func-test-leap15: false
Skip-func-hw-test: true
Skip-scan-rpms: true
Skip-test-rpms: true
Test-tag: test_core_files
Skip-PR-comments: true

brianjmurrell committed:
Fix import


#Pragmas from previous commit message:
Quick-Functional: true
PR-repos: daos@PR-11447:15
RPM-test-version: 2.3.103
Skip-coverity-test: true
Skip-func-test-leap15: false
Skip-func-hw-test: true
Skip-scan-rpms: true
Skip-test-rpms: true
Test-tag: test_core_files
Skip-PR-comments: true
Skip-checkpatch: true
Skip-python-bandit: true
Skip-build: true
Quick-build: true
Quick-Functional: true
Skip-unit-tests: true
Skip-test: true
Allow-unstable-test: true
#RPM-test-version: version[-release]
#RPM-test-version: 2.3.103
# VM1-label: ci_vm1
# Ubuntu-VM9-label: ci_vm9
# Leap15-VM9-label: ci_vm9
# EL8-VM9-label: ci_vm9
Quick-Functional: true
Skip-coverity-test: true
Skip-func-test-leap15: false
Skip-func-hw-test: true
Skip-scan-rpms: true
Skip-test-rpms: true
Test-tag: test_core_files
Skip-PR-comments: true
Quick-Functional: true
Skip-coverity-test: true
Skip-func-test-leap15: false
Skip-func-hw-test: true
Skip-scan-rpms: true
Skip-test-rpms: true
Test-tag: test_core_files
Skip-PR-comments: true
Quick-Functional: true
Skip-coverity-test: true
Skip-func-test-leap15: false
Skip-func-hw-test: true
Skip-scan-rpms: true
Skip-test-rpms: true
Test-tag: test_core_files
Skip-PR-comments: true
Signed-off-by: Brian J. Murrell <brian.murrell@intel.com>
Quick-Functional: true
Skip-coverity-test: true
Skip-func-test-leap15: false
Skip-func-hw-test: true
Skip-scan-rpms: true
Skip-test-rpms: true
Test-tag: test_core_files
Skip-PR-comments: true
EL8-VM9-label: stage_vm9

brianjmurrell committed:
Updates


#Pragmas from previous commit message:
Quick-Functional: true
PR-repos: daos@PR-11447:15
RPM-test-version: 2.3.103
Skip-coverity-test: true
Skip-func-test-leap15: false
Skip-func-hw-test: true
Skip-scan-rpms: true
Skip-test-rpms: true
Test-tag: test_core_files
Skip-PR-comments: true
Skip-checkpatch: true
Skip-python-bandit: true
Skip-build: true
Quick-build: true
Quick-Functional: true
Skip-unit-tests: true
Skip-test: true
Allow-unstable-test: true
#RPM-test-version: version[-release]
#RPM-test-version: 2.3.103
# VM1-label: ci_vm1
# Ubuntu-VM9-label: ci_vm9
# Leap15-VM9-label: ci_vm9
# EL8-VM9-label: ci_vm9
Quick-Functional: true
Skip-coverity-test: true
Skip-func-test-leap15: false
Skip-func-hw-test: true
Skip-scan-rpms: true
Skip-test-rpms: true
Test-tag: test_core_files
Skip-PR-comments: true
Quick-Functional: true
Skip-coverity-test: true
Skip-func-test-leap15: false
Skip-func-hw-test: true
Skip-scan-rpms: true
Skip-test-rpms: true
Test-tag: test_core_files
Skip-PR-comments: true
Quick-Functional: true
Skip-coverity-test: true
Skip-func-test-leap15: false
Skip-func-hw-test: true
Skip-scan-rpms: true
Skip-test-rpms: true
Test-tag: test_core_files
Skip-PR-comments: true
Quick-Functional: true
Skip-coverity-test: true
Skip-func-test-leap15: false
Skip-func-hw-test: true
Skip-scan-rpms: true
Skip-test-rpms: true
Test-tag: test_core_files
Skip-PR-comments: true
EL8-VM9-label: stage_vm9
Signed-off-by: Brian J. Murrell <brian.murrell@intel.com>
Quick-Functional: true
Skip-coverity-test: true
Skip-func-test-leap15: false
Skip-func-hw-test: true
Skip-scan-rpms: true
Skip-test-rpms: true
Test-tag: test_core_files
Skip-PR-comments: true
EL8-VM9-label: stage_vm9

brianjmurrell committed:
Updates

Quick-Functional: true
Skip-coverity-test: true
Skip-func-test-leap15: false
Skip-func-hw-test: true
Skip-scan-rpms: true
Skip-test-rpms: true
Test-tag: test_core_files
Skip-PR-comments: true
EL8-VM9-label: stage_vm9

Required-githooks: true

Signed-off-by: Brian J. Murrell <brian.murrell@intel.com>

phender commented on src/tests/ftest/launch.py:
```diff
@@ -601,6 +600,8 @@ def __init__(self, name, mode):
         self.slurm_control_node = NodeSet()
         self.slurm_partition_hosts = NodeSet()
         self.slurm_add_partition = False
+        self.core_file_processing = CoreFileProcessing(logger)
```
To avoid the import issue with the avocado's settings, let's do this:
```suggestion
        self.core_file_processing = None
```

brianjmurrell replied:
Indeed.  That was the part of the solution I had already figured out.  Where to set it was what I had to do next.

brianjmurrell commented on src/tests/ftest/launch.py:
```diff
@@ -2500,7 +2462,8 @@ def _archive_files(self, summary, hosts, source, pattern, destination, depth, th
         if test and "core files" in summary:
             # Process the core files
-            return_code |= self._process_core_files(os.path.split(destination)[0], test)
+            return_code |= self._process_core_files(os.path.split(destination)[0] + os.path.sep,
+                                                    test)
```
It's not so much a case of _needed_ but more just a case about being explicit that a target is a directory.  It prevents cases such as successfully doing a `cp foo bar` where `bar/` doesn't exist yet and you carete a file instead of [failing] copying into a [missing] directory.

brianjmurrell commented on src/tests/ftest/launch.py:
```diff
@@ -2218,11 +2165,26 @@ def process(self, test, repeat, stop_daos, archive, rename, jenkinslog, core_fil
-                    "source": core_files[hosts]["path"],
+
+            # Get the core file pattern information
+            if not process_cores:
+                logger.debug("Not collecting core files")
+                return
+
+            try:
+                self.core_files = self.core_file_processing.get_core_file_pattern(
+                    test.host_info.servers.hosts | test.host_info.clients.hosts,
+                    self.result.tests[-1].time_start)
+            except CoreFileException:
+                message = "Error obtaining the core file pattern information"
+                self._fail_test(self.result.tests[-1], "Process", message, sys.exc_info())
+                return_code |= 256
```
I made it a class member as it's also needed on L2720.  I'm open to alternatives.

brianjmurrell committed:
Updates

Signed-off-by: Brian J. Murrell <brian.murrell@intel.com>
Quick-Functional: true
Skip-coverity-test: true
Skip-func-test-leap15: false
Skip-func-hw-test: true
Skip-scan-rpms: true
Skip-test-rpms: true
Test-tag: test_core_files
Skip-PR-comments: true
EL8-VM9-label: stage_vm9

phender commented on src/tests/ftest/util/core_file.py:
```diff
@@ -0,0 +1,420 @@
+
+        self.log = log
+        self.distro_info = detect()
+
```
Let's  get the local core file path here:
```suggestion
        host = get_local_host()
        result = self.get_core_file_pattern(host)
        self.local_core_path = result[str(host)])['path']
```

brianjmurrell replied:
Done.

phender commented on src/tests/ftest/util/core_file.py:
```diff
@@ -0,0 +1,420 @@
+
+        return core_files
+
+    def process_core_files(self, directory, delete, core_path=None, test=None):
```
```suggestion
    def process_core_files(self, directory, delete, test=None):
```

brianjmurrell replied:
Done.

phender commented on src/tests/ftest/util/core_file.py:
```diff
@@ -0,0 +1,420 @@
+                        core_file = os.path.join(core_dir, core_name)
+                        self.log.debug("Removing %s", core_file)
+                        os.remove(core_file)
+        if core_path:
+            # remove any core file generated post core processing on the local node
+            errors += self.delete_gdb_core_files(core_path)
```
```suggestion
        # remove any core file generated post core processing on the local node
        errors += self.delete_gdb_core_files(self.local_core_path)
```

phender commented on src/tests/ftest/launch.py:
```diff
@@ -601,6 +600,8 @@ def __init__(self, name, mode):
         self.slurm_partition_hosts = NodeSet()
         self.slurm_add_partition = False
+        self.core_file_processing = None
+        self.core_files = {}
```
```suggestion
```

brianjmurrell replied:
Done.

phender commented on src/tests/ftest/launch.py:
```diff
@@ -2218,15 +2165,31 @@ def process(self, test, repeat, stop_daos, archive, rename, jenkinslog, core_fil
+            if process_cores:
+                try:
+                    self.core_file_processing = CoreFileProcessing(logger)
+                    self.core_files = self.core_file_processing.get_core_file_pattern(
```
```suggestion
                    core_files = self.core_file_processing.get_core_file_pattern(
```

brianjmurrell replied:
Done.

phender commented on src/tests/ftest/launch.py:
```diff
@@ -2218,15 +2165,31 @@ def process(self, test, repeat, stop_daos, archive, rename, jenkinslog, core_fil
+                except CoreFileException:
+                    message = "Error obtaining the core file pattern information"
+                    self._fail_test(self.result.tests[-1], "Process", message, sys.exc_info())
+                    return_code |= 256
```
```suggestion
                    return_code |= 256
                    core_files = {}
```

brianjmurrell replied:
Done.

phender commented on src/tests/ftest/launch.py:
```diff
@@ -2218,15 +2165,31 @@ def process(self, test, repeat, stop_daos, archive, rename, jenkinslog, core_fil
+                    self._fail_test(self.result.tests[-1], "Process", message, sys.exc_info())
+                    return_code |= 256
+
+                for index, hosts in enumerate(self.core_files):
```
```suggestion
                for index, hosts in enumerate(core_files):
```

brianjmurrell replied:
Done.

phender commented on src/tests/ftest/launch.py:
```diff
@@ -2749,10 +2714,13 @@ def _process_core_files(self, test_job_results, test):
+            corefiles_processed = self.core_file_processing.process_core_files(
+                test_job_results,
+                True,
+                [self.core_files[k]['path'] for k in list(self.core_files.keys())
+                    if NodeSet(self.local_host) in NodeSet(k)][0],
```
```suggestion
```

brianjmurrell commented on src/tests/ftest/harness/core_files.py:
```diff
@@ -41,13 +42,10 @@ def test_core_files(self):
-            self.fail("Unable to find local core file pattern")
-        core_path = os.path.split(results.stdout.splitlines()[-1])[0]
-        core_file = "{}/core.gdb.harness.advanced".format(core_path)
+        host = get_local_host()
+        result = CoreFileProcessing(self.log).get_core_file_pattern(host)
 
+        core_file = "{}/core.gdb.harness.advanced".format((result[str(host)])['path'])
```
Done.

brianjmurrell commented on src/tests/ftest/launch.py:
```diff
@@ -2218,15 +2165,31 @@ def process(self, test, repeat, stop_daos, archive, rename, jenkinslog, core_fil
+                try:
+                    self.core_file_processing = CoreFileProcessing(logger)
+                    self.core_files = self.core_file_processing.get_core_file_pattern(
+                        test.host_info.servers.hosts | test.host_info.clients.hosts,
```
Done.

brianjmurrell commented on src/tests/ftest/util/core_file.py:
```diff
@@ -0,0 +1,420 @@
+
+        Args:
+            all_hosts (NodeSet): hosts used in testing
+
```
Done.

brianjmurrell committed:
Updates per Phil's review

Signed-off-by: Brian J. Murrell <brian.murrell@intel.com>
Quick-Functional: true
Skip-coverity-test: true
Skip-func-test-leap15: false
Skip-func-hw-test: true
Skip-scan-rpms: true
Skip-test-rpms: true
Test-tag: test_core_files
Skip-PR-comments: true
EL8-VM9-label: stage_vm9

brianjmurrell committed:
Merge remote-tracking branch 'origin/master' into bmurrell/default-to-saving-coredumps

Signed-off-by: Brian J. Murrell <brian.murrell@intel.com>

brianjmurrell committed:
Fix pylint error

Signed-off-by: Brian J. Murrell <brian.murrell@intel.com>
Skip-PR-comments: true
Features: test_core_files

daosbuild1 commented on src/tests/ftest/scripts/get_core_file_pattern.sh:
```diff
@@ -0,0 +1,28 @@
+# systemd-coredump core path
+if [[ "$core_pattern" = *systemd-coredump* ]]; then
+    # move the core files where our processing scripts wants to find them
+    coredumpctl list -S @"$1" --no-legend | while read -r d dd t tz pid _uid _gid _sig _status exe; do
```
(style)  line over 100 characters

daosbuild1 commented on src/tests/ftest/scripts/get_core_file_pattern.sh:
```diff
@@ -0,0 +1,28 @@
+        if [[ $exe = */gdb ]]; then
+            continue
+        fi
+        coredumpctl dump --output /var/tmp/core."${exe##*/}"."$(date +%s -d"$d $dd $t $tz")"."$pid" "$pid"
```
(style)  line over 100 characters

daosbuild1 reviewed (dismissed):
Style warning(s) for job https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-11447/24/
Please review https://wiki.hpdd.intel.com/display/DC/Coding+Rules

brianjmurrell committed:
Fix linting issues

Skip-PR-comments: true
Test-tag: pr test_core_files
Skip-func-test-leap15: false
Skip-func-hw-test: true
Skip-unit-tests: true

Signed-off-by: Brian J. Murrell <brian.murrell@intel.com>

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

brianjmurrell commented:
Could I please get another review on this from any of the requested reviewers (you are likely one if you are getting this notification) other than @raydennis who has already reviewed.

It's going to need a master merge as it is, but I'd rather get the reviewing done before pushing an update to this again.

brianjmurrell committed:
Merge remote-tracking branch 'origin/master' into bmurrell/default-to-saving-coredumps

Test-tag: pr test_core_files
Skip-func-test-leap15: false
Skip-func-hw-test: true
Skip-unit-tests: true

Signed-off-by: Brian J. Murrell <brian.murrell@intel.com>
Required-githooks: true

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

brianjmurrell commented:
@JohnMalmberg @raydennis @phender Unfortunately needs a re-review as it bit-rotted until it was conflicting with current master and the conflict resolutions cleared previous review flags.  Would appreciate your earliest opportunity on this round to prevent that from happening again.  P&T.

daosbuild1 commented:
Test stage Functional on Leap 15.4 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-11447/26/execution/node/856/log

daosbuild1 commented:
Test stage Functional on EL 8 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-11447/26/execution/node/866/log

brianjmurrell committed:
core_file_processing is a class member now

Test-tag: pr test_core_files
Skip-func-test-leap15: false
Skip-func-hw-test: true
Skip-unit-tests: true

Required-githooks: true

Signed-off-by: Brian J. Murrell <brian.murrell@intel.com>

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

brianjmurrell committed:
Merge remote-tracking branch 'origin/master' into bmurrell/default-to-saving-coredumps

Test-tag: pr test_core_files
Skip-func-test-leap15: false
Skip-func-hw-test: true
Skip-unit-tests: true

Required-githooks: true

Signed-off-by: Brian J. Murrell <brian.murrell@intel.com>

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

brianjmurrell commented:
Once again, re-review is required due to needing to merge yet again due to conflicts.  Could I please request timely reviews on this PR so that we are not doing this again in a few days ad infinitum?

brianjmurrell commented:
It's now over a month later and this PR is still waiting for reviews.  Amazingly it does not seem to need any kind of master merge which is entirely shocking given that it modifies the `debian/changelog` as well as `daos.spec` files.

It's really a shame that this has not yet landed as it has literally more than a month's worth of work invested in it.

It's doesn't seem to be *required* for 2.4 but it does have _Fix Versions_ set for 2.4 and 2.2.1.
